File: client/android/app/src/main/java/app/mobiling/client/auth/AuthFeatureBridge.kt
package app.mobiling.client.auth

import app.mobiling.client.contract.auth.session.AuthSessionPayload
import app.mobiling.client.contract.auth.session.ConfirmVerificationRequest
import app.mobiling.client.contract.auth.session.RegisterAuthRequest
import app.mobiling.client.contract.auth.session.StartAuthRequest
import app.mobiling.client.contract.auth.session.VerifySecondFactorRequest
import app.mobiling.client.data.auth.session.AuthSessionGateway
import app.mobiling.client.usecase.auth.session.ChallengeSecondFactorUseCase
import app.mobiling.client.usecase.auth.session.ConfirmVerificationUseCase
import app.mobiling.client.usecase.auth.session.LogoutAuthUseCase
import app.mobiling.client.usecase.auth.session.RegisterAuthUseCase
import app.mobiling.client.usecase.auth.session.ResendVerificationUseCase
import app.mobiling.client.usecase.auth.session.RestoreAuthUseCase
import app.mobiling.client.usecase.auth.session.StartAuthUseCase
import app.mobiling.client.usecase.auth.session.VerifySecondFactorUseCase

/**
 * Marketing America Corp. Oleksandr Tishchenko
 *
 * App-level bridge for auth entry flow.
 *
 * It keeps Auth as the dedicated entry/auth surface while delegating
 * behavior into canonical Contract/Data/UseCase slices.
 */
class AuthFeatureBridge(
    private val gateway: AuthSessionGateway,
) {
    suspend fun start(request: StartAuthRequest): AuthSessionPayload =
        StartAuthUseCase(gateway).invoke(request)

    suspend fun register(request: RegisterAuthRequest): AuthSessionPayload =
        RegisterAuthUseCase(gateway).invoke(request)

    suspend fun restore(): AuthSessionPayload =
        RestoreAuthUseCase(gateway).invoke()

    suspend fun logout() =
        LogoutAuthUseCase(gateway).invoke()

    suspend fun resendVerification(): AuthSessionPayload =
        ResendVerificationUseCase(gateway).invoke()

    suspend fun confirmVerification(request: ConfirmVerificationRequest): AuthSessionPayload =
        ConfirmVerificationUseCase(gateway).invoke(request)

    suspend fun challengeSecondFactor(): AuthSessionPayload =
        ChallengeSecondFactorUseCase(gateway).invoke()

    suspend fun verifySecondFactor(request: VerifySecondFactorRequest): AuthSessionPayload =
        VerifySecondFactorUseCase(gateway).invoke(request)
}

File: client/ios/App/Auth/AuthFeatureBridge.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
// App-level bridge for auth entry flow.
public struct AuthFeatureBridge: Sendable {
    private let gateway: AuthSessionGateway

    public init(gateway: AuthSessionGateway) {
        self.gateway = gateway
    }

    public func start(request: StartAuthRequest) async throws -> AuthSessionPayload {
        try await StartAuthUseCase(gateway: gateway)(request: request)
    }

    public func register(request: RegisterAuthRequest) async throws -> AuthSessionPayload {
        try await RegisterAuthUseCase(gateway: gateway)(request: request)
    }

    public func restore() async throws -> AuthSessionPayload {
        try await RestoreAuthUseCase(gateway: gateway)()
    }

    public func logout() async throws {
        try await LogoutAuthUseCase(gateway: gateway)()
    }

    public func resendVerification() async throws -> AuthSessionPayload {
        try await ResendVerificationUseCase(gateway: gateway)()
    }

    public func confirmVerification(request: ConfirmVerificationRequest) async throws -> AuthSessionPayload {
        try await ConfirmVerificationUseCase(gateway: gateway)(request: request)
    }

    public func challengeSecondFactor() async throws -> AuthSessionPayload {
        try await ChallengeSecondFactorUseCase(gateway: gateway)()
    }

    public func verifySecondFactor(request: VerifySecondFactorRequest) async throws -> AuthSessionPayload {
        try await VerifySecondFactorUseCase(gateway: gateway)(request: request)
    }
}